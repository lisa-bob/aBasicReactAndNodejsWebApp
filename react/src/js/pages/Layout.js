import React from "react";

// Elemente von React Router einbinden
import { HashRouter, Route } from 'react-router-dom';

// Komponenten einbinden
import NavigationBar from "../components/NavigationBar";

// verschiedene Seiten (pages) einbinden
import userState from "./userState";
import Books from "../pages/Books";
import MoreInteraction from "./MoreInteraction";

export default class Layout extends React.Component {
    
    render() {

        return (
            <HashRouter>
                <div>
                    <NavigationBar location={location}/>
                    <div class="container">
                        <div class="row">
                            <div class="col-xs-12">
                            <Route exact path="/user" component={userState}/>
                            <Route exact path="/login" component={userState}/>
                            <Route exact path="/moreInteraction" component={MoreInteraction}/>
                            <Route exact path="/" component={userState}/>
                            <Route exact path="/books" component={Books}/>
                            </div>
                        </div>
                    </div>
                </div>
            </HashRouter>
        );
    }
}

/*




         
                                
                                

*/