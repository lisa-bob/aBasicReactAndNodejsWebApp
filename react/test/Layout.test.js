import React from "react"
import Layout from "../src/js/pages/Layout"
import welcomeTextStore from "../src/js/stores/welcomeTextStore"

test('Render Layout', () => {
    const wrapper = shallow(
        <Layout welcomeTextStore={welcomeTextStore}/>
    );

    expect(wrapper).toMatchSnapshot();
});
