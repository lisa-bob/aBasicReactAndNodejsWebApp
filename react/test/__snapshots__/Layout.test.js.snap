// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Layout 1`] = `
<HashRouter>
  <div>
    <NavigationBar />
    <div
      className="container"
      style={
        Object {
          "marginTop": "60px",
        }
      }
    >
      <div
        className="row"
      >
        <div
          className="col-xs-12"
        >
          <h1>
            Hello World
          </h1>
          <button
            className="btn btn-primary"
            onClick={[Function]}
          >
            Fetch another text
          </button>
          <Route
            component={[Function]}
            exact={true}
            path="/index"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/initial"
          />
        </div>
      </div>
    </div>
  </div>
</HashRouter>
`;
